src/jit/hwintrinsicxarch.cpp

```diff
@@ -203,10 +246,8 @@ GenTree* Compiler::impX86HWIntrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HA
     InstructionSet isa = isaOfHWIntrinsic(intrinsic);
     // Will throw PlatformNotSupportedException if
     // - calling hardware intrinsics on unsupported hardware
-    // - calling SIMD hardware intrinsics with featureSIMD=false
-    if ((!compSupports(isa) || (!featureSIMD && isa != InstructionSet_BMI1 && isa != InstructionSet_BMI2 &&
-                                isa != InstructionSet_LZCNT && isa != InstructionSet_POPCNT)) &&
-        !isIntrinsicAnIsSupportedPropertyGetter(intrinsic))
+    // - JIT does not support this hardware intrinsics (compSupportHWIntrinsic return false)
+    if ((!compSupports(isa) || !compSupportHWIntrinsic(isa)) && !isIntrinsicAnIsSupportedPropertyGetter(intrinsic))
```

Comment:
Could you extract the check for the specific ISA intrinsics at the beginning, e.g.
```
bool supported = (featureSIMD && compSupports(isa) && compSupportsHWIntrinsics(isa));
```
Then, the above check can be:
```
if (!supported && !isIntrinsicAnIsSupportedPropertyGetter(intrinsic))
```
and the checks below can be:
```
retNode = gtNewIconNode(supported);
```

Reply:
Good point, will do.